e2e-test/src/test/java/io/cdap/plugin/normalize/stepsdesign/NormalizeTransform.java

```diff
@@ -0,0 +1,128 @@
+  }
+
+  public static void verifyPluginPropertyInvalidInlineErrorMessage(String propertyName,
+                                                                   String expectedInlineErrorMessage) {
+    WaitHelper.waitForElementToBeDisplayed(NormalizeLocators.locatePluginPropertyInvalidInlineError(propertyName));
+    AssertionHelper.verifyElementDisplayed(NormalizeLocators.locatePluginPropertyInvalidInlineError(propertyName));
+    AssertionHelper.verifyElementContainsText(NormalizeLocators.locatePluginPropertyInvalidInlineError(propertyName),
+                                              expectedInlineErrorMessage);
+  }
+
+  public static void verifyPluginPropertyInvalidInlineErrorMessageColor(String propertyName) {
+    logger.info("Verify if plugin property: " + propertyName + "'s inline error message is shown in the expected color:"
+                  + " " + "#a40403");
+    String actualColor = ElementHelper.getElementColorCssProperty(NormalizeLocators.
+      locatePluginPropertyInvalidInlineError(propertyName));
+    String expectedColor = "#a40403";
+    Assert.assertEquals(expectedColor, actualColor);
+  }
+
+  @Then("Select Normalize plugin property: output schema with value: {string}")
+  public void selectNormalizePluginPropertyOutputSchemaWithValue(String action) throws InterruptedException {
+      NormalizeActions.selectClearOutputSchema(PluginPropertyUtils.pluginProp(action));
+    }
+
+  @Then("Enter Normalize plugin FieldsMapping {string}")
+  public void enterNormalizePluginFieldsMapping(String jsonFieldsMapping) {
+    NormalizeActions.enterFieldsMapping(jsonFieldsMapping);
+  }
+
+  @Then("Enter outputSchema {string}")
+  public void enterOutputSchema(String jsonOutputSchema) {
+    NormalizeActions.enterOutputSchema(jsonOutputSchema);
+  }
+
+  @Then("Enter Normalize plugin FieldsNormalize {string}")
+  public void enterNormalizePluginFieldsNormalize(String jsonFieldsNormalize) {
+    NormalizeActions.enterFieldsToBeNormalized(jsonFieldsNormalize);
+  }
+
+  @Then("Validate OUT record count is equal to records transferred to target BigQuery table")
+  public void validateOUTRecordCountIsEqualToRecordsTransferredToTargetBigQueryTable()
+          throws IOException, InterruptedException, IOException {
+    int targetBQRecordsCount = BigQueryClient.countBqQuery(PluginPropertyUtils.pluginProp("bqTargetTable"));
+    BeforeActions.scenario.write("No of Records Transferred to BigQuery:" + targetBQRecordsCount);
+    Assert.assertEquals("Out records should match with target BigQuery table records count",
+            CdfPipelineRunAction.getCountDisplayedOnSourcePluginAsRecordsOut(), targetBQRecordsCount);
+  }
+
+  @Then("Validate OUT record count of transpose is equal to IN record count of sink")
+  public void validateOUTRecordCountOfTransposeIsEqualToINRecordCountOfSink() {
+    Assert.assertEquals((long) this.recordOutCount(), (long) this.recordInCount());
+  }
+
+  public int recordInCount() {
```

Comment:
remove recordInCount and recordOutCount from this file -> Add single method in action to validate